src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -203,38 +221,86 @@ private void LaunchDevelopmentProxy()
+                WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+            };
 
-                    var stopProcess = Process.Start(stopScriptInfo);
-                    if (stopProcess == null || stopProcess.HasExited)
-                    {
-                        _logger.LogWarning($"SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
-                            $" remain open if the dotnet process is terminated abruptly. Use the operating system command to kill" +
-                            $"the process tree for {_spaProcess.Id}");
-                    }
-                    else
-                    {
-                        _logger.LogDebug($"Watch process '{stopProcess}' started.");
-                    }
-                }
+            var stopProcess = Process.Start(stopScriptInfo);
+            if (stopProcess == null || stopProcess.HasExited)
+            {
+                _logger.LogWarning($"The SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
+                    $" remain open if the dotnet process is terminated ungracefully. Use the operating system commands to kill" +
+                    $" the process tree for {_spaProcess!.Id}");
             }
-            catch (Exception exception)
+            else
             {
-                _logger.LogError(exception, $"Failed to launch the SPA development server '{_options.LaunchCommand}'.");
+                _logger.LogDebug($"Watch process '{stopProcess}' started.");
             }
         }
 
+        private void LaunchStopScriptMacOS()
+        {
+            var fileName = Guid.NewGuid().ToString("N") + ".sh";
+            var scriptPath = Path.Combine(AppContext.BaseDirectory, fileName);
+            var stopScript = @$"function list_child_processes(){{
+    local ppid=$1;
+    local current_children=$(pgrep -P $ppid);
+    local local_child;
+    if [ $? -eq 0 ];
+    then
+        for current_child in $current_children
+        do
+          local_child=$current_child;
+          list_child_processes $local_child;
+          echo $local_child;
+        done;
+    else
+      return 0;
+    fi;
+}}
+
+ps {Environment.ProcessId};
+while [ $? -eq 0 ];
+do
+  sleep 1;
+  ps {Environment.ProcessId} > /dev/null;
+done;
+
+for child in $(list_child_processes {_spaProcess!.Id});
+do
+  echo killing $child;
+  kill -s KILL $child;
+done;
+rm {scriptPath};
+";
+            File.WriteAllText(scriptPath, stopScript);
+
+            var stopScriptInfo = new ProcessStartInfo(
+                "/bin/bash",
+                string.Join(" ", scriptPath))
+            {
+                CreateNoWindow = true,
+                WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+            };
+
+            var stopProcess = Process.Start(stopScriptInfo);
+            if (stopProcess == null || stopProcess.HasExited)
```

Comment:
Are we waiting for the process to run to completion here? If not it seems strange to be checking for `stopProcess.HasExited`, as that would randomly be true or false depending on how quickly it happens to complete.

Reply:
It's checking that it doesn't exit immediately, which happens if the command is incorrect.

Reply:
Does that definitely happen synchronously?

Reply:
Yes, the typical case for this is that something is wrong in the setup and either the process doesn't get created at all or the command executed but fails instantly. The worst outcome of this is a potential missing warning, if we wanted to be more accurate we can attach to the exit event handler and do it there, but I don't think its worth the additional complexity